src/Parser/TypeParser.php

```diff
@@ -858,6 +859,19 @@ private function parseArrayShape(TokenIterator $tokens, Ast\Type\TypeNode $type,
 
 			if ($tokens->tryConsumeTokenType(Lexer::TOKEN_VARIADIC)) {
 				$sealed = false;
+				$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
+				if ($tokens->isCurrentTokenType(Lexer::TOKEN_OPEN_ANGLE_BRACKET)) {
+					$extraItemType = $this->parseGeneric(
```

Comment:
I thought about adjusting the grammar, but there currently doesn't seem to be a big focus on keeping the grammar maintained. There's e.g. no representation for `TOKEN_PHPDOC_EOL` at all, `ArrayShape` is already out of date etc. 
I could work on all of that, but again, if there's not much interest in keeping it maintained that doesn't quite seem worthwhile.

Ultimately this leads to the question what the point of the grammar is. 
Is it mainly for documentation / easier understanding of the parser logic? Or is there a goal to generate more documentation based on it?
Is the end goal to move to a generated parser?
Should it serve for third party consumers?

I suppose ideally the tests would somehow verify that the parse results match the current grammar.